src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -134,6 +134,23 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
             case 2:
             {
+                if (((intrinsicId == NI_SSE_Store) || (intrinsicId == NI_SSE2_Store)) && op2->isContained())
```

Comment:
Feel free to ignore if you aren't ready for comments. Was just wondering why put this here, rather than as a special-case under `if (category == HW_Category_MemoryStore)` on https://github.com/dotnet/coreclr/pull/22896/files#diff-847442c7efdf7a78e78711bea30fd4b2R173?

Reply:
Right, it seems that it should work. I wasn't quite sure that `genConsumeOperands` will handle this case.

Anyway, waiting for @CarolEidt 's containment PR, I see that she also added a special case here, again because the `genConsumeOperands` it seems.